project_cost_report/static/src/js/report_action.js

```diff
@@ -0,0 +1,386 @@
+        'click .o_project_cost_report__product_total': 'productTotalClicked',
+        'click .o_project_cost_report__time_category': 'timeCategoryClicked',
+        'click .o_project_cost_report__time_total_hours': 'hourTotalClicked',
+        'click .o_project_cost_report__time_total': 'hourTotalClicked',
+        'click .o_project_cost_report__outsourcing_category': 'outsourcingCategoryClicked',
+        'click .o_project_cost_report__outsourcing_total': 'outsourcingTotalClicked',
+    },
+    init(parent, action) {
+        this._super.apply(this, arguments);
+        this.controllerURL = action.context.url;
+        this.projectId = action.context.active_id || action.params.active_id;
+        this.reportContext = {
+            active_id: this.projectId,
+            unfolded_categories: {
+                'product': [],
+                'time': [],
+                'outsourcing': [],
+            },
+        };
+        this.unfolded = false;
+    },
+    /**
+     * Update the HTML content of the report.
+     */
+    async updateHtml(){
+        var html = await this._rpc({
+            model: 'project.cost.report',
+            method: 'get_html',
+            args: [this.reportContext],
+        });
+        this.$el.html(html);
+    },
+    /**
+     * Render the PRINT button.
+     *
+     * returns {jQuery} the button node
+     */
+    renderPrintButton(){
+        var button = $(QWeb.render("projectCostReport.printButton", {}));
+        button.bind('click', () => this.downloadPDF());
+        return button;
+    },
+    /**
+     * Render the FOLD button.
+     *
+     * returns {jQuery} the button node
+     */
+    renderFoldButton(){
+        var button = $(QWeb.render("projectCostReport.foldButton", {}));
+        button.bind('click', () => this.fold());
+        return button;
+    },
+    /**
+     * Render the UNFOLD button.
+     *
+     * returns {jQuery} the button node
+     */
+    renderUnfoldButton(){
+        var button = $(QWeb.render("projectCostReport.unfoldButton", {}));
+        button.bind('click', () => this.unfold());
+        return button;
+    },
+    /**
+     * Render the buttons of the control panel.
+     *
+     * returns {Array<jQuery>} an array of button nodes
+     */
+    getControlPanelButtons(){
+        if(!this.controlPanelButtons){
+            this.printButton = this.renderPrintButton();
+            this.foldButton = this.renderFoldButton();
+            this.unfoldButton = this.renderUnfoldButton();
+            this.controlPanelButtons = [this.printButton, this.foldButton, this.unfoldButton];
+        }
+        return this.controlPanelButtons;
+    },
+    /**
+     * Update the control pannel of the report view.
+     */
+    updateControlPanel(){
+        this.update_control_panel({
+            breadcrumbs: this.getParent().get_breadcrumbs(),
+            cp_content: {$buttons: this.getControlPanelButtons()},
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Identifier 'cp_content' is not in camel case. (camelcase)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=3675949)